### PR TITLE
Deactivate the DEBUG logs and provide resources to jar

### PR DESCRIPTION
**Pull request by BadrTad**
Provide logback.xml to configure logger

### CODE REVIEW

File: be2-scala/src/main/resources/logback.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+<configuration debug="false" scan="true" scanPeriod="15 seconds">
+  <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+    <encoder>
+      <pattern>%d{HH:mm:ss.SSS} [%thread] %-5level %logger{36} - %msg%n</pattern>
+    </encoder>
+  </appender>
+
+<!--Set root(general) logs level to INFO/DEBUG on STDOUT-->
+  <root level="INFO">
+    <appender-ref ref="STDOUT"/>
```
Comment (pierluca) [inline be2-scala/src/main/resources/logback.xml:12]: This would be a separate PR, but if you want to take advantage of [Apache Chainsaw](https://logging.apache.org/chainsaw/2.x/), you could configure a socket appender.
It's really super helpful when debugging locally.

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/Validator.scala [hunk 2]
```diff
@@ -60,7 +59,8 @@ object Validator {
     val factory: JsonSchemaFactory = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V7)
     // Creation of a JsonNode using the readTree function from the file query.json (at queryPath)
     // FIXME: error handling for queryPath
-    val jsonSchemaNode: JsonNode = objectMapper.readTree(new File(queryPath))
+    val jsonSchemaNode: JsonNode = objectMapper.readTree(queryFile)
+    queryFile.close()
```
Comment (pierluca) [inline be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/Validator.scala:63]: won't this lead to a failure the next time `validateSchema` is called? 

Also, for readability, it might make sense to create queryFile right before it's used, use it, and close it afterwards.

File: be2-scala/build.sbt [hunk 3]
```diff
@@ -1,11 +1,40 @@
...
+
+//Create task to copy the protocol folder to resources
+lazy val copyProtocolTask = taskKey[Unit]("Copy protocol to resources")
+copyProtocolTask := {
+    val log = streams.value.log
+    log.info("Executing Protocol folder copy...")
+    val scalaDest = "be2-scala"
+    baseDirectory.value.name
+    if(! baseDirectory.value.name.equals(scalaDest)){
+        log.error(s"Please make sure you working dir is $scalaDest !")
```
Comment (pierluca) [inline be2-scala/build.sbt:21]: very minor point, but is there no way to do away with this ? can't we take the `build.sbt` location as a reference ?
below there are many other files that seem to be referenced with relative paths.

**Review by pierluca — COMMENTED** [review]
Had a quick look, looks nice !
I have a few minor questions, but I'll let @RaulinN do the proper review :-)

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/Validator.scala (hunk 2 above)
Comment (BadrTad) [reply be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/Validator.scala]: No it does not fail....however it's unnecessary as the `objectMapper.readTree` does the closing.
Additionally, the current state of the method is computationally heavy, meaning to validate every message the validator has to read and process json skeleton from `query.json` (constant) to validate the `jsonString`.  
Calls to readTree are expensive. A fix to this was made in commit `9a6be31fad59817fa369969d26efc60c3217e759` to do this in a nice way. This is a major issue as I have repeatedly noticed a delay of the responses. now the delay is only noticeable in the first message all the others are instant.
A improvement could be computing the schema at compile time for good performance.

File: be2-scala/build.sbt (hunk 3 above)
Comment (BadrTad) [reply be2-scala/build.sbt:21]: Assuming this:
 ..
 |___protocol
 |___be-scala
       |____src/main/resource/protocol
       |____build.sbt
 The project "be2-scala" should be open in IDE such that `build.sbt` is at the root of the workspace and the working directory is `be2-scala` . If not `sbt `does not recognize the build at all.
 
Plus there is no need to worry about this since it's done at compile time. once the jar is properly assembled.

File: be2-scala/src/main/resources/logback.xml (hunk 1 above)
Comment (BadrTad) [reply be2-scala/src/main/resources/logback.xml:12]: 👍

**Review by BadrTad — COMMENTED** [review]
Careful consideration of `readTree` calls

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/Validator.scala (hunk 2 above)
Comment (pierluca) [reply be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/Validator.scala]: Thank you for the nice analysis, it seems your new solution is much better indeed (though I haven't looked into all the tiny details) ! 

As opposed to compile-time evaluation, it could also be done on the startup of the server (before the evaluation of the first message), but don't bother with it. There's no need to push the optimization that far ;-)

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/Validator.scala [hunk 4]
```diff
@@ -31,44 +31,33 @@ import scala.util.{Failure, Success, Try}
...
-      case Success(Seq(JsNumber(id))) => Some(id.toInt)
-      case _ => None
-    }
-  }
-
-  // contains the relative path to our "main" query JsonSchema file
-  final val queryPath = "../protocol/query/query.json"
-
-  def validateSchema(jsonString: JsonString): Either[JsonString, PipelineError] = {
+  private final val queryPath = "protocol/query/query.json" //With respect to resource folder
```
Comment (tuomas-p) [inline be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/Validator.scala:39]: looks good

File: be2-scala/src/main/scala/ch/epfl/pop/config/application.conf [hunk 5]
```diff
@@ -1,8 +1,21 @@
...
+akka {
+  #Setup list off loggers to config
+  loggers = ["akka.event.slf4j.Slf4jLogger"]
+  #Setup
+  logging-filter = "akka.event.slf4j.Slf4jLoggingFilter"
+  # Options: OFF, ERROR, WARNING, INFO, DEBUG
+  loglevel = "INFO"
+  #Set to OFF to turn off logs for sys startup & shuttdown logs
+  stdout-loglevel = "INFO"
+}
```
Comment (tuomas-p) [inline be2-scala/src/main/scala/ch/epfl/pop/config/application.conf:17]: :+1:

File: be2-scala/.gitignore [hunk 6]
```diff
@@ -19,3 +19,6 @@ database
 
 # Test
 database_test
+
+#Protocol folder
+src/main/resources/protocol
```
Comment (tuomas-p) [inline be2-scala/.gitignore:24]: :+1:

**Review by tuomas-p — APPROVED** [review]
Looks very good to me!

File: be2-scala/src/main/resources/logback.xml [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  </root>
```
Comment (RaulinN) [inline be2-scala/src/main/resources/logback.xml:13]: So this was the secret! I remember seeing this for the SimpleLogger we looked at. Good job making it work with that!

File: be2-scala/build.sbt [hunk 8]
```diff
@@ -1,11 +1,40 @@
+import scala.util.{Try, Success, Failure}
```
Comment (RaulinN) [inline be2-scala/build.sbt:1]: Seems to work (copyProtocolTask) as expected

**Review by RaulinN — APPROVED** [review]
